src/client/array/dc_array.c

```diff
@@ -1064,14 +1092,291 @@ create_sgl(d_sg_list_t *user_sgl, daos_size_t cell_size,
 	return 0;
 }
 
+struct hole_params {
+	struct io_params	*io_list;
+	tse_task_t		*ptask;
+	daos_size_t		records_req;
+	daos_size_t		array_size;
+	daos_handle_t		oh;
+};
+
+static int
+memset_sgl(uint8_t *buf, size_t len, void *args)
```

Comment:
Minor nit: reset_buf?   memset implies being able to specify what to set it to and sgl seems irrelevant here.

Reply:
OK, will wait for more comments from others before refreshing.